### PR TITLE
Added {Def,Reset}{Prog,Shell}Mode() functions

### PR DESCRIPTION
**Pull request by xaionaro**
Fixing https://github.com/rthornton128/goncurses/issues/41

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,10 +8,12 @@ Installation
 The go tool is the recommended method of installing goncurses. Issue the
 following command on the command line:
 
-$ go get github.com/rthornton128/goncurses
+```sh
+go get github.com/xaionaro-go/goncurses
```
Comment (rthornton128) [inline README.md]: This line should be reverted `github.com/rthornton128/goncurses`. The rest of the MR looks fine.

Comment (xaionaro) [reply README.md]: Fixed.

**Comment by xaionaro** [comment]
@limar63 : Sorry, I do not remember why I added that method and how did I use it :(

**Comment by limar63** [comment]
You mean you don't remember what is the purpose of it? I wanted to use the method to have no delay on getting Getch(), combined with Nap() so I could make a good control for my snake game on ncurses (Halfdelay isn't working properly because it is instantly applied after getting user input without time delay).

**Comment by rthornton128** [comment]
On review, I think this needs to be split into two separate PRs. One to add the Debian build scripts and a second to add the new/missing methods.

**Comment by rthornton128** [comment]
Closing due to inactivity.

**Comment by xaionaro** [comment]
> @limar63 : Sorry, I do not remember why I added that method and how did I use it :(

Yeah, sorry. Two years between doing that and getting a review -- considering how many code got through me during that period I already forgot what this was for.

**Pull request closed without merging**